### PR TITLE
optimize the favicon.ico image

### PR DESCRIPTION
**Pull request by peterbe**
Part of #3611

I used a program called Optimage on my mac which produced this new file. 

Here the two images are zoomed in **4x** next to each other. Yes, at 4x you can tell that they are different, but can you really tell which one is the optimized one?
Screen Shot 2021-04-23 at 12 38 58 PM

### CODE REVIEW

**Comment by schalkneethling** [comment]
I think one other question here is do we even need an `ico` seeing that we already specify others and `ico` is really the ultimate fallback? I did quite a bit of reading into all of this:
https://150daysofhtml.substack.com/p/the-link-element-part3

I would appreciate your thoughts here. Thanks for digging into this @peterbe

**Comment by peterbe** [comment]
> I think one other question here is do we even need an `ico` seeing that we already specify others and `ico` is really the ultimate fallback? I did quite a bit of reading into all of this:
> https://150daysofhtml.substack.com/p/the-link-element-part3
> 
What does it say? :)

> I would appreciate your thoughts here. Thanks for digging into this @peterbe


It does not need to be called `favicon.ico` but that exact name is a convention. If you load a URL that isn't HTML you'll find that the browser also loads the `/favicon.ico` for that domain. E.g. 
Screen Shot 2021-04-26 at 10 47 41 AM
It shows up nicely in the Firefox dev tools. But not in Chrome's:
Screen Shot 2021-04-26 at 10 48 30 AM

But as you can see, in Chrome, it still fetched a favicon to be used on that tab. How is it supposed to know which favicon if there's no HTML loaded? The answer is the default convention of `/favicon.ico`. 

What a lot of sites have is both. On GitHub for example, they control the favicon (green checkmark or red dot) from the HTML (and possibly a web API too) so if you wanted to you can use a PNG. And keep the `/favicon.ico` as a fallback for browsers that need it and don't read HTML. 
I think on Squarespace sites they use a 100x100 JPEG. E.g. view-source:https://www.dedeandbv.com/

In conclusion, if you think we can get an even better favicon image as PNG (or JPEG) then I'm happy to change to that and keep the old `/favicon.ico` purely as a fallback. So if you can beat the size (and quality) of the one I prepared here, we can easily switch to that instead.

**Comment by peterbe** [comment]
So in https://github.com/mdn/yari/pull/3621 we change it so that for almost all browsers they get the super optimized (48x48) PNG. But, when the HTML isn't loaded the browser might still fall back on the `/favicon.ico` and it would thus be nice if that image is as small as it possibly can be. 
So I think this PR is good to go. 

@schalkneethling r?

**Review by schalkneethling — APPROVED** [review]
